src/dill/gauge-proxy-v2.sol

```diff
@@ -0,0 +1,2161 @@
+                        tokenLastVotedPeriodId[_token]
+                    ][_token];
+
+                    tokenLastVotedPeriodId[_token] = _currentId;
+                }
+
+                weights[_currentId][_token] += _tokenWeight;
+                votes[_owner][_token] = _tokenWeight;
+                tokenVote[_owner].push(_token);
+
+                if (_tokenWeight < 0) _tokenWeight = -_tokenWeight;
+
+                _usedWeight += _tokenWeight;
+                totalWeight[_currentId] += _tokenWeight;
+            }
+        }
+        usedWeights[_owner] = _usedWeight;
+    }
+
+    // Vote with DILL on a gauge
+    function vote(address[] calldata _tokenVote, int256[] calldata _weights)
+        external
+    {
+        require(
+            _tokenVote.length == _weights.length,
+            "GaugeProxy: token votes count does not match weights count"
+        );
+        uint256 currentId = getCurrentPeriodId();
+        require(currentId > 0, "Voting not started yet");
+        _vote(msg.sender, _tokenVote, _weights, currentId);
+        delegations[msg.sender].blockDelegate[currentId] = true;
+    }
+
+    function setVotingDelegate(
+        address _delegateAddress,
+        uint256 _periodsCount,
+        bool _indefinite
+    ) external {
+        require(
+            _delegateAddress != address(0),
+            "GaugeProxyV2: cannot delegate zero address"
+        );
+        require(
+            _delegateAddress != msg.sender,
+            "GaugeProxyV2: delegate address cannot be delegating"
+        );
+
+        delegateData storage _delegate = delegations[msg.sender];
+
+        uint256 currentPeriodId = getCurrentPeriodId();
+
+        address currentDelegate = _delegate.delegate;
+        _delegate.delegate = _delegateAddress;
+        _delegate.prevDelegate = currentDelegate;
+        _delegate.updatePeriodId = currentPeriodId;
+
+        if (_indefinite == true) {
+            _delegate.indefinite = true;
+        } else if (_delegate.prevDelegate == address(0)) {
+            _delegate.endPeriod = currentPeriodId + _periodsCount - 1;
+        } else {
+            _delegate.endPeriod = currentPeriodId + _periodsCount;
+        }
+    }
+
+    function voteFor(
+        address _owner,
+        address[] calldata _tokenVote,
+        int256[] calldata _weights
+    ) external {
+        require(
+            _tokenVote.length == _weights.length,
+            "GaugeProxy: token votes count does not match weights count"
+        );
+
+        uint256 currentId = getCurrentPeriodId();
+        require(currentId > 0, "Voting not started yet");
+        delegateData storage _delegate = delegations[_owner];
+        require(
+            (_delegate.delegate == msg.sender &&
+                currentId > _delegate.updatePeriodId) ||
+                (_delegate.prevDelegate == msg.sender &&
+                    currentId == _delegate.updatePeriodId) ||
+                (_delegate.prevDelegate == address(0) &&
```

Comment:
Doesn't this condition make it so that if the user previously had no delegate and decided to delegate this period, that ANYONE would be able to vote on their behalf?

Reply:
(_delegate.prevDelegate == address(0) && currentId == _delegate.updatePeriodId

The second check in the && condition ensures that this will work only if a delegate is set. If not set _delegate.updatePeriodId is 0 and currentId is never 0. If previous delegate is set, this OR condition is not checked as it passes for the previous OR conditions.